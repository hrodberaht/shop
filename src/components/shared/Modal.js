import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';

export default class Modal extends Component {
  static propTypes = {
    handleCancelClick: PropTypes.func.isRequired,
    handleConfirmClick: PropTypes.func.isRequired,
  };

  render() {
    const { handleConfirmClick, handleCancelClick } = this.props;
    return ReactDOM.createPortal(
      <div className="modal">
        <div className="inner-modal">
          <h2>Are you sure?</h2>
          <div className="inner-modal__buttons">
            <button type="button" className="btn btn-submit" onClick={handleCancelClick}>
              Cancel
            </button>
            <button type="button" className="btn btn-danger" onClick={handleConfirmClick}>
              Confirm
            </button>
          </div>
        </div>
      </div>,
      document.body,
    );
  }
}
