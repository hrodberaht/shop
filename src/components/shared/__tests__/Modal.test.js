import Modal from '../Modal';

const defaultProps = {
  handleCancelClick: jest.fn(),
  handleConfirmClick: jest.fn(),
};

const setup = buildSetup(Modal, defaultProps);

describe('<Modal />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleCancelClick after click on cancel button', () => {
    const { wrapper } = setup();
    wrapper
      .find('button')
      .at(0)
      .simulate('click');
    expect(wrapper.instance().props.handleCancelClick).toHaveBeenCalled();
  });
  it('should call handleConfirmClick after click on confirm button', () => {
    const { wrapper } = setup();
    wrapper
      .find('button')
      .at(1)
      .simulate('click');
    expect(wrapper.instance().props.handleConfirmClick).toHaveBeenCalled();
  });
});
