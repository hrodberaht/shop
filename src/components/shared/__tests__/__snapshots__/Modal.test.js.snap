// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render without crash 1`] = `
<Portal
  containerInfo={<body />}
>
  <div
    className="modal"
  >
    <div
      className="inner-modal"
    >
      <h2>
        Are you sure?
      </h2>
      <div
        className="inner-modal__buttons"
      >
        <button
          className="btn btn-submit"
          onClick={[MockFunction]}
          type="button"
        >
          Cancel
        </button>
        <button
          className="btn btn-danger"
          onClick={[MockFunction]}
          type="button"
        >
          Confirm
        </button>
      </div>
    </div>
  </div>
</Portal>
`;
